src/test/java/org/broadinstitute/hellbender/tools/walkers/haplotypecaller/HaplotypeCallerIntegrationTest.java

```diff
@@ -571,10 +577,127 @@ public void testMinBaseQualityScore() throws Exception {
 
     }
 
+    @DataProvider
+    public Object[][] getContaminationTestData() {
+        // This bam is a snippet of GATKBaseTest.NA12878_20_21_WGS_bam artificially contaminated
+        // at 15% with reads from another sample. We use such a high contamination fraction to ensure
+        // that calls will be noticeably different vs. running without -contamination at all.
+        final String contaminatedBam15Percent = largeFileTestDir + "contaminated_bams/CEUTrio.HiSeq.WGS.b37.NA12878.CONTAMINATED.WITH.HCC1143.NORMALS.AT.15PERCENT.bam";
+        final SimpleInterval traversalInterval = new SimpleInterval("20", 10000000, 10250000);
+
+        // bam,
+        // known contamination fraction,
+        // interval,
+        // reference,
+        // GVCF mode,
+        // calls on the uncontaminated bam
+        // expected concordance vs. the calls on the uncontaminated bam
+        return new Object[][] {
+                { contaminatedBam15Percent,
+                  0.15,
+                  traversalInterval,
+                  b37_reference_20_21,
+                  false, // VCF mode
+                  TEST_FILES_DIR + "expected.CEUTrio.HiSeq.WGS.b37.NA12878.calls.20.10000000-10250000.vcf",
+                  // Expect > 90% concordance vs. uncontaminated calls
```

Comment:
I'm curious - why would it not be ~85% without the contamination argument given that the test data are 15% contaminated?

Reply:
Well, 15% of the reads come from another sample. Depending on how they are distributed, they could easily end up affecting more than (or less than) 15% of the variant calls.

Reply:
Ah, OK.  I understand now.  Thanks.